mlkem/src/native/riscv64/src/rv64v_poly.c

```diff
@@ -0,0 +1,811 @@
+  MLK_RVV_BFLY_RX(v1, v3, vt, izeta[0x20], vl);
+  MLK_RVV_BFLY_RX(v4, v6, vt, izeta[0x21], vl);
+  MLK_RVV_BFLY_RX(v5, v7, vt, izeta[0x21], vl);
+  MLK_RVV_BFLY_RX(v8, va, vt, izeta[0x30], vl);
+  MLK_RVV_BFLY_RX(v9, vb, vt, izeta[0x30], vl);
+  MLK_RVV_BFLY_RX(vc, ve, vt, izeta[0x31], vl);
+  MLK_RVV_BFLY_RX(vd, vf, vt, izeta[0x31], vl);
+
+  MLK_RVV_BFLY_RX(v0, v4, vt, izeta[0x10], vl);
+  MLK_RVV_BFLY_RX(v1, v5, vt, izeta[0x10], vl);
+  MLK_RVV_BFLY_RX(v2, v6, vt, izeta[0x10], vl);
+  MLK_RVV_BFLY_RX(v3, v7, vt, izeta[0x10], vl);
+  MLK_RVV_BFLY_RX(v8, vc, vt, izeta[0x11], vl);
+  MLK_RVV_BFLY_RX(v9, vd, vt, izeta[0x11], vl);
+  MLK_RVV_BFLY_RX(va, ve, vt, izeta[0x11], vl);
+  MLK_RVV_BFLY_RX(vb, vf, vt, izeta[0x11], vl);
+
+  MLK_RVV_BFLY_RX(v0, v8, vt, izeta[0x01], vl);
+  MLK_RVV_BFLY_RX(v1, v9, vt, izeta[0x01], vl);
+  MLK_RVV_BFLY_RX(v2, va, vt, izeta[0x01], vl);
+  MLK_RVV_BFLY_RX(v3, vb, vt, izeta[0x01], vl);
+  MLK_RVV_BFLY_RX(v4, vc, vt, izeta[0x01], vl);
+  MLK_RVV_BFLY_RX(v5, vd, vt, izeta[0x01], vl);
+  MLK_RVV_BFLY_RX(v6, ve, vt, izeta[0x01], vl);
+  MLK_RVV_BFLY_RX(v7, vf, vt, izeta[0x01], vl);
+
+  v0 = fq_mulq_vx(v0, MLK_MONT_NR, vl);
+  v1 = fq_mulq_vx(v1, MLK_MONT_NR, vl);
+  v2 = fq_mulq_vx(v2, MLK_MONT_NR, vl);
+  v3 = fq_mulq_vx(v3, MLK_MONT_NR, vl);
+  v4 = fq_mulq_vx(v4, MLK_MONT_NR, vl);
+  v5 = fq_mulq_vx(v5, MLK_MONT_NR, vl);
+  v6 = fq_mulq_vx(v6, MLK_MONT_NR, vl);
+  v7 = fq_mulq_vx(v7, MLK_MONT_NR, vl);
+  v8 = fq_mulq_vx(v8, MLK_MONT_NR, vl);
+  v9 = fq_mulq_vx(v9, MLK_MONT_NR, vl);
+  va = fq_mulq_vx(va, MLK_MONT_NR, vl);
+  vb = fq_mulq_vx(vb, MLK_MONT_NR, vl);
+  vc = fq_mulq_vx(vc, MLK_MONT_NR, vl);
+  vd = fq_mulq_vx(vd, MLK_MONT_NR, vl);
+  ve = fq_mulq_vx(ve, MLK_MONT_NR, vl);
+  vf = fq_mulq_vx(vf, MLK_MONT_NR, vl);
+
+  __riscv_vse16_v_i16m1(&r[0x00], v0, vl);
+  __riscv_vse16_v_i16m1(&r[0x10], v1, vl);
+  __riscv_vse16_v_i16m1(&r[0x20], v2, vl);
+  __riscv_vse16_v_i16m1(&r[0x30], v3, vl);
+  __riscv_vse16_v_i16m1(&r[0x40], v4, vl);
+  __riscv_vse16_v_i16m1(&r[0x50], v5, vl);
+  __riscv_vse16_v_i16m1(&r[0x60], v6, vl);
+  __riscv_vse16_v_i16m1(&r[0x70], v7, vl);
+  __riscv_vse16_v_i16m1(&r[0x80], v8, vl);
+  __riscv_vse16_v_i16m1(&r[0x90], v9, vl);
+  __riscv_vse16_v_i16m1(&r[0xa0], va, vl);
+  __riscv_vse16_v_i16m1(&r[0xb0], vb, vl);
+  __riscv_vse16_v_i16m1(&r[0xc0], vc, vl);
+  __riscv_vse16_v_i16m1(&r[0xd0], vd, vl);
+  __riscv_vse16_v_i16m1(&r[0xe0], ve, vl);
+  __riscv_vse16_v_i16m1(&r[0xf0], vf, vl);
+}
+
+#undef MLK_RVV_BFLY_RX
+#undef MLK_RVV_BFLY_RV
+
+/*  Kyber's middle field GF(3329)[X]/(X^2) multiplication   */
+
+static inline void mlk_rv64v_poly_basemul_mont_add_k(int16_t *r,
+                                                     const int16_t *a,
+                                                     const int16_t *b,
+                                                     unsigned kn)
+{
+  /* check-magic: off */
+  const int16_t roots[MLKEM_N] = {
```

Comment:
Those should be autogenerated by `autogen`

Reply:
I implemented autogeneration in 559a7fad7aebf105693f0673b56d8f945d9dd8c4